zebra/zebra_vxlan.c

```diff
@@ -5914,18 +5918,28 @@ static int zebra_vxlan_sg_send(struct zebra_vrf *zvrf,
 static unsigned int zebra_vxlan_sg_hash_key_make(const void *p)
 {
 	const struct zebra_vxlan_sg *vxlan_sg = p;
+	uint32_t hash1;
+	uint32_t hash2;
 
-	return (jhash_2words(vxlan_sg->sg.src.s_addr,
-				vxlan_sg->sg.grp.s_addr, 0));
+	if (IS_IPADDR_V4(&vxlan_sg->sg.src)) {
+		return (jhash_2words(vxlan_sg->sg.src.ipaddr_v4.s_addr,
+				     vxlan_sg->sg.grp.s_addr, 0));
+	} else {
+		hash1 = jhash_1word(vxlan_sg->sg.grp.s_addr, 0);
+		hash2 = jhash2(vxlan_sg->sg.src.ipaddr_v6.s6_addr32,
+			       array_size(vxlan_sg->sg.src.ipaddr_v6.s6_addr32),
+			       0);
+		return jhash_2words(hash1, hash2, 0);
```

Comment:
Why can't we put the hash1 value into the jhash2 call on line 5929 and just return?

Reply:
Good point, done